python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
Thanks @jerryshao , tool change-scala-version.sh is what I needed.

After building with scala-2.10 without redundant code， spark-shell works well.
So I removed them and updated the pull request.